### DISCUSSION TITLE
Treating tab as one character in new file created via `:n`

### DISCUSSION THREAD

**Discussion by nuid64** [helix-editor/helix#8560] [General] [answered]
### Summary

After creation of a new file via `:n` and saving it as a some file with supported extension tab is treated as a whole character and not as a bunch of spaces.
This fixes after `:set-language <lang>` and only if there is no such tabs in the file.

### Reproduction Steps

I tried this:

1. `hx`
2. '/n'
3. `:w foo.c`
4. <Tab>foo

I expected this to happen:
A bunch of spaces before `foo` occured

Instead, this happened:
There was a tab

### Platform

Linux

### Terminal Emulator

alacritty 0.12.3 (5efb0690)

### Helix Version

helix 23.05 (18522924)

**Comment by the-mikedavis** [comment, marked as answer]
This is intentional, see https://github.com/helix-editor/helix/pull/3095
